internal/backend/cache/backend.go

```diff
@@ -215,3 +218,40 @@ func (b *Backend) IsNotExist(err error) bool {
 func (b *Backend) Unwrap() backend.Backend {
 	return b.Backend
 }
+
+func (b *Backend) List(ctx context.Context, t backend.FileType, fn func(f backend.FileInfo) error) error {
```

Comment:
This change makes the `err := r.Cache.Clear(restic.IndexFile, indexIDs)` call in `internal/repository/repository.go` redundant.